lib/reversetunnel/agentpool.go

```diff
@@ -271,13 +271,16 @@ func (p *AgentPool) run() error {
 		if err != nil {
 			if p.ctx.Err() != nil {
 				return nil
-			} else if isProxyAlreadyClaimed(err) {
+			}
+
+			level := slog.LevelWarn
+			if isProxyAlreadyClaimed(err) && !p.IsRemoteCluster {
 				// "proxy already claimed" is a fairly benign error, we should not
 				// spam the log with stack traces for it
-				p.logger.DebugContext(p.ctx, "Failed to connect agent", "error", err)
-			} else {
-				p.logger.DebugContext(p.ctx, "Failed to connect agent", "error", err)
+				level = slog.LevelDebug
```

Comment:
To actually match what the comment says:

```suggestion
				level = slog.LevelDebug
				err = trace.Unwrap(err)
```